src/Maestro/Maestro.MergePolicies/MergePolicy.cs

```diff
@@ -47,6 +49,41 @@ public string Name
 
         public abstract string DisplayName { get; }
 
-        public abstract Task EvaluateAsync(IMergePolicyEvaluationContext context, MergePolicyProperties properties);
+        public abstract Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc);
+
+        public Task<MergePolicyEvaluationResult> Pending(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Pending, message, this));
+        }
+
+        public Task<MergePolicyEvaluationResult> Succeed(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Success, message, this));
+        }
+
+        public Task<MergePolicyEvaluationResult> Fail(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Failure, message, this));
+        }
+    }
+
+    public interface IMergePolicy : IMergePolicyInfo
+    {
+        Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc);
+    }
+
+    public interface IMergePolicyBuilder
+    {
+        public string Name { get; }
```

Comment:
This isn't valid code... the fact that the build completed is... troublesome.  Interface members can't have access modifiers...

Reply:
Even if it is, it's... highly unusual (GitHub doesn't even understand enough to highlight it), so we should probably remove it.

Reply:
Good catch. R# would tell, but ... :-(